addons/sevenrings/script_component.hpp

```diff
@@ -0,0 +1,50 @@
+#define COMPONENT seven_rings
+#define COMPONENT_BEAUTIFIED Seven Rings
+#include "\x\tacu\addons\main\script_mod.hpp"
+#include "\x\tacu\addons\main\script_macros.hpp"
+
+#define MACRO_SCOPE \
```

Comment:
Maybe `MACRO_SCOPE_HIDE` or something? I'd at least consider clarifying that this macro sets the scope to 1 / hides stuff.